components/skus/browser/rs/cxx/src/lib.rs

```diff
@@ -465,3 +499,35 @@ async fn credential_summary_task(
         Err(e) => callback.0(callback_state.into_raw(), e, ""),
     }
 }
+
+#[repr(transparent)]
+pub struct SubmitReceiptCallback(
+    pub extern "C" fn(
+        callback_state: *mut ffi::SubmitReceiptCallbackState,
+        result: ffi::SkusResult,
+        summary: &str,
```

Comment:
I'm confused why we're passing a `&str` back since the success return type of `submit_receipt` is `()`?

Reply:
https://github.com/brave/brave-core/pull/13820/commits/7c4a876b15517f43eb604e307ab0c509c4407bd0